### ISSUE TITLE
[bug] False positive assigning built-in function to instance attribute with built-in function type

### ISSUE THREAD

**Issue by alexchandel** [python/mypy#9049]
(1) Invoking a built-in function assigned to a class attribute via instance attribute access, or (2) assigning a built-in function to an instance attribute that would shadow such a class attribute, like the `logging`-recommended way to change the time format,

```python
import logging, time
fmt = logging.Formatter
fmt.converter(42.0)  # (1)
fmt.converter = time.gmtime  # (2)
```

yields the following false positive in mypy in both cases:

```
Invalid self argument "Formatter" to attribute function "converter" with type "Callable[[Optional[float]], struct_time]"
```

mypy correctly resolved the original `fmt.converter` to `Formatter.converter`, and correctly resolved that to `time.localtime` with signature `Callable[[Optional[float]], struct_time]`.  However `localtime` is a built-in function, meaning `fmt.converter` is **not** a bound method.  Similarly for `gmtime`.

**Comment by ilevkivskyi** [comment]
I think this is a duplicate of https://github.com/python/mypy/issues/2427
